contain-intrinsic-size.bs

```diff
@@ -0,0 +1,125 @@
+          applies.  In effect, this feature extends [=size containment=] by 
+          letting the developer specify the placeholder size to apply when the
+          element's children are not considered.
+</pre>
+
+Introduction {#intro}
+============
+
+When [=size containment=] is specified, the element does not consider its
+children when determining its intrinsic size, which in turn can affect the
+element's [=principal box=] size. This is benefitial for performance, as it can
+allow the user-agent to skip processing layout of elements in the size
+contained subtrees. However, it also makes it difficult to create responsive
+design content, since the element needs to be sized for it to take up layout
+space.
+
+The proposed property, 'contain-intrinsic-size', allows the developer to
+specify a size that replaces the intrinsic size when [=size containment=] is
+specified. This means that [=size containment=] can be toggled on and off, with
+the element either considering the specified value or its children for
+intrinsic sizing. This is particularly helpful for features such as <a
+href="https://wicg.github.io/display-locking/#subtree-visibility-auto">
+'subtree-visibility: auto'</a>, which applies [=size containment=]
+automatically based on element's distance to the visual viewport.
+
+The 'contain-intrinsic-size' property {#contain-intrinsic-size}
+=====================================
+
+<pre class=propdef>
+Name: contain-intrinsic-size
+Value: auto | <<length>>{1,2}
+Initial: auto
+Inherited: no
+Applies to: elements with [=size containment=]
+</pre>
+
+<dl dfn-type=value dfn-for=contain-intrinsic-size>
+    : <dfn>auto</dfn>
+    :: No effect; the element is sized as usual.
+
+    : <<length>>{1,2}
+    :: If the element is not under [=size containment=], this value has no
+         effect.
+
+         Otherwise,
+             * The first <<length>> specifies the [=min-content size=] and
+                 the [=max-content size=] for the [=physical=] width direction
+                 of the element's [=principal box=].
+             * The second <<length>> specifies the
+                 [=min-content size=] and the [=max-content size=] for the
+                 [=physical=] height direction of the element's [=principal
+                 box=]. 
+
+         If one <<length>> is provided, the second length defaults to the same
+         value as the first one.
+</dl>
+
+Examples {#examples}
+========
+
+<div class=example>
+  <xmp highlight=markup>
+    <style>
+    #one {
+      contain-intrinsic-size: 10px 20px;
+      contain: size;
+    }
+    #two {
+      contain-intrinsic-size: 10px 20px;
+    }
+    #three {
+      contain-intrinsic-size: 50px;
+      contain: size;
+      width: max-content;
+    }
+    #four {
+      contain-intrinsic-size: 50px;
+      contain: size;
+      width: 27px;
+      height: 42px;
+    }
+    </style>
+
+    <div id=one></div>
```

Comment:
Add descendant element of each of these?

Reply:
Sounds good.